src/processor/common.ts

```diff
@@ -0,0 +1,16 @@
+import { ExportableAssignmentRule, ExportableSdRule } from '../exportable';
+
+// Places general Quantity-setting rules ahead of Quantity.unit setting rules
+export function switchQuantityRules(rules: ExportableSdRule[]): void {
+  const seenRules = new Map();
+  rules.forEach((rule, index) => {
+    if (!seenRules.has(rule.path) && rule instanceof ExportableAssignmentRule)
```

Comment:
I think we would also want this to work on `ExportableCaretValueRule`s. It is possible that this situation could arise there as well. It's a bit of a pain, since on the `CaretValueRule` we care about the `caretPath` and not the `path`, but there is a `getRulePath` function [here](https://github.com/FHIR/GoFSH/blob/ad7c7f8b4b279a1ff1ecd4ee19cc2d877aef8bab/src/optimizer/plugins/ConstructInlineInstanceOptimizer.ts#L117) that you could probably use to make that easier.

Reply:
You're right about that! I'll add support for `ExportableCaretValueRule`s as well. Would the `caretPath` matter for what we're trying to accomplish though? I'm thinking that only the regular path would matter here even in the case of an `ExportableCaretValueRule`, since that's the portion that specifies which element is being set. I may be wrong here though!

Reply:
I was wrong, and took your suggestion haha. Thank you!